### DISCUSSION TITLE
Metamask and Metamask Flask conflicting

### DISCUSSION THREAD

**Discussion by rmuftiev** [MetaMask/snaps#721] [General]
HI there,
just my finding as fresh Snap builder, that Metamask and Metamask Flask conflicting when installed together and I try to run Snap.
I uninstalled regular Metmask and it works.  It is more as finding for others here. 

FYI, a bit more details: 
macOS Monterrey 12.5 
Browser: Chrome, Opera, Firefox

<img width="555" alt="error1" src="https://user-images.githubusercontent.com/37123778/187101356-35f37cd4-08f1-4879-943d-0cca906986c4.png">
<img width="281" alt="error2" src="https://user-images.githubusercontent.com/37123778/187101358-f738cfc4-3710-40b6-a8ad-8b02b03b7f24.png">

In pic,   nkbihfbeogaeaoehlefnkodbefgpgknn   is address of my Metamask
ljfoeinjpaedjfecbmggjgodbgkmjkjk is address of my Metamask Flask

**Comment by noomly** [comment]
Afaik this is a known issue due to the way `window.ethereum` is injected. The only workaround I'm aware of atm if you're developing a snap/dapp is to add a timeout to the requests you're sending to Metamask and prompt your user to verify they are not running two instances of Metamask if the request expires.

**Comment by Montoya** [comment]
Please do not run MetaMask and MetaMask Flask in the same browser profile. You can set up a separate Chrome profile for Flask and install it separately.
